sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +310,102 @@ export class ManagementClient extends LinkEntity {
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
   ): Promise<any> {
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            if (!options) {
```

Comment:
This check is already happening in the outer function scope. We could get rid of both places this is checked and use a default parameter assignment in _makeManagementRequest instead... that should let us stop using the TypeScript non-null assertion as often too.

Reply:
Discussed offline about setting default value continuing to treat parameter as optional - updated!